scylla/src/transport/session.rs

```diff
@@ -128,14 +128,22 @@ impl Session {
     pub async fn query(
         &self,
         query: impl Into<Query>,
-        values: &[Value],
+        values: &Result<SerializedValues, AddValueError>,
```

Comment:
I think this function should not take `&Result<SerializedValues, AddValueError>` and return immediately if it holds `Err` value.

Reply:
If I understand it correctly and this parameter is virtually always a result of calling our `scylla::values` macro, then it's alright for me. Perhaps we could typedef it to look nicer (`type QueryValuesInput = (...)`).
@piodul opinions?